aw_client/client.py

```diff
@@ -445,18 +452,44 @@ def should_stop(self) -> bool:
     def _dispatch_request(self) -> None:
         request = self._get_next()
         if not request:
-            self.wait(0.1)  # seconds to wait before re-polling the empty queue
+            self.wait(0.2)  # seconds to wait before re-polling the empty queue
             return
 
         try:
             self.client._post(request.endpoint, request.data)
-        except req.RequestException as e:
+        except req.exceptions.ConnectTimeout:
+            # Triggered by:
+            #   - server not running (connection refused)
+            #   - server not responding (timeout)
+            # Safe to retry according to requests docs:
+            #   https://requests.readthedocs.io/en/latest/api/#requests.ConnectTimeout
+
             self.connected = False
             logger.warning(
-                "Failed to send request to aw-server, will queue requests until connection is available."
+                "Connection refused or timeout, will queue requests until connection is available."
             )
+            # wait a bit before retrying, so we don't spam the server (or logs), see:
+            #  - https://github.com/ActivityWatch/activitywatch/issues/815
+            #  - https://github.com/ActivityWatch/activitywatch/issues/756#issuecomment-1266662861
+            sleep(0.5)
             return
+        except req.RequestException as e:
+            if e.response.status_code == 400:
```

Comment:
`e.response` might be None

Reply:
Made a PR to typeshed to help discover bug like this with mypy: https://github.com/python/typeshed/pull/8989